openfpga/src/fpga_bitstream/build_grid_bitstream.cpp

```diff
@@ -230,11 +232,34 @@ static void build_physical_block_pin_interc_bitstream(
       const PhysicalPbId& des_pb_id =
         physical_pb.find_pb(des_pb_graph_pin->parent_node);
       size_t mux_input_pin_id = 0;
-      if (true != physical_pb.valid_pb_id(des_pb_id)) {
+      if (true != physical_pb.valid_pb_id(des_pb_id)) { /* Unmapped pb */
         mux_input_pin_id = DEFAULT_PATH_ID;
-      } else if (AtomNetId::INVALID() == physical_pb.pb_graph_pin_atom_net(
+      } else if (AtomNetId::INVALID() == physical_pb.pb_graph_pin_atom_net( /* Unmapped output */
                                            des_pb_id, des_pb_graph_pin)) {
-        mux_input_pin_id = DEFAULT_PATH_ID;
+        if(false == circuit_lib.mux_add_const_input(mux_model) && fix_unmapped_mux_selection){
```

Comment:
I just realize that you do not need to add similar codes to build_grid_bitstream.cpp, build_routing_bitstream.cpp. 
You may just add the codes to the function which decodes all the MUX bitstream:

https://github.com/lnis-uofu/OpenFPGA/blob/c76a2cde6560383962881e02825ed33512abb8fa/openfpga/src/fpga_bitstream/build_mux_bitstream.cpp#L78-L84

Reply:
I think `find_mux_default_path_id` doesn't have access to some data structures we use, and it could complicate that function if we try to pass them into there

Reply:
My bad. Current codes are good enough. Thanks!